packages/gitbook/src/components/AIActions/AIActions.tsx

```diff
@@ -0,0 +1,214 @@
+'use client';
+
+import { useAIChatController } from '@/components/AI/useAIChat';
+import { useAIChatState } from '@/components/AI/useAIChat';
+import { ChatGPTIcon } from '@/components/AIActions/assets/ChatGPTIcon';
+import { ClaudeIcon } from '@/components/AIActions/assets/ClaudeIcon';
+import { MarkdownIcon } from '@/components/AIActions/assets/MarkdownIcon';
+import AIChatIcon from '@/components/AIChat/AIChatIcon';
+import { Button } from '@/components/primitives/Button';
+import { DropdownMenuItem } from '@/components/primitives/DropdownMenu';
+import { tString, useLanguage } from '@/intl/client';
+import type { TranslationLanguage } from '@/intl/translations';
+import { Icon, type IconName, IconStyle } from '@gitbook/icons';
+import { useState } from 'react';
+
+type AIActionType = 'button' | 'dropdown-menu-item';
+
+export function OpenDocsAssistant(props: { type: AIActionType }) {
+    const { type } = props;
+    const chatController = useAIChatController();
+    const chat = useAIChatState();
+    const language = useLanguage();
+
+    return (
+        <AIActionWrapper
+            type={type}
+            icon={
+                chat.loading ? (
+                    <Icon icon="spinner-third" className="size-3.5 animate-spin" />
+                ) : (
+                    <AIChatIcon />
+                )
+            }
```

Comment:
```suggestion
            icon={<AIChatIcon state={chat.loading ? 'thinking' : 'default'} />}
```
I'll update the icon soon, this way it'll work automatically 😊